gnosis/eth/utils.py

```diff
@@ -4,9 +4,97 @@
 import eth_abi
 from eth._utils.address import generate_contract_address
 from eth_keys import keys
-from eth_utils import to_canonical_address, to_checksum_address
+from eth_typing import AnyAddress, ChecksumAddress, HexStr
+from eth_utils import to_normalized_address
 from hexbytes import HexBytes
-from web3 import Web3
+from sha3 import keccak_256
+
+
+def fast_keccak(value: bytes) -> bytes:
+    """
+    Calculates ethereum keccak256 using fast library `pysha3`
+    :param value:
+    :return: Keccak256 used by ethereum as `bytes`
+    """
+    return keccak_256(value).digest()
+
+
+def fast_keccak_hex(value: bytes) -> HexStr:
+    """
+    Same as `fast_keccak`, but it's a little more optimal calling `hexdigest()`
+    than calling `digest()` and then `hex()`
+
+    :param value:
+    :return: Keccak256 used by ethereum as an hex string (not 0x prefixed)
+    """
+    return HexStr(keccak_256(value).hexdigest())
+
+
+def _build_checksum_address(
+    norm_address: HexStr, address_hash: HexStr
+) -> ChecksumAddress:
+    """
+    :param norm_address: address in lowercase (not 0x prefixed)
+    :param address_hash: keccak256 of `norm_address` (not 0x prefixed)
+    :return:
+    """
+    return ChecksumAddress(
+        "0x"
+        + (
+            "".join(
+                (
+                    norm_address[i].upper()
+                    if int(address_hash[i], 16) > 7
+                    else norm_address[i]
```

Comment:
I'd add a link to https://github.com/ethereum/EIPs/blob/master/EIPS/eip-55.md as a comment

Reply:
Nice, done @mikhailxyz